### ISSUE TITLE
Keypoints Matching for Multiple People during Evaluation 

### ISSUE THREAD

**Issue by thribhuvanrapolu** [shubham-goel/4D-Humans#130]
During evaluation, if we have multiple people in a frame how does the code make sure that ground and predicted 2d/3d keypoints are for the same person?

I have been working on 3DPW dataset. Lets say the model pred_2d_keypoints is of order person 2,1,3,4 but the ground_2d_keypoints is of order 1,2,3,4 
but in the code (4D-Humans/hmr2/utils/pose_utils.py) its directly calculating the error without considering this case 

`
kp_err = torch.nn.functional.mse_loss(
                        pred_keypoints_2d,
                        gt_keypoints_2d[:, :, :, :-1],
                        reduction='none'
                    ).sum(dim=3)
`

**Comment by geopavlakos** [comment]
During evaluation, we use as input the ground truth bounding box of the person that we want to evaluate, so there is no confusion.
We have released the evaluation code for 3DPW, so you can see how we perform the evaluation.

**Comment by thribhuvanrapolu** [comment]
I have come across hmr2_evaluation_data folder while trying to understand evaluation part.

In that the 3DPW dataset doesn't have ground truth bounding box. So where exactly are we getting the ground truth bounding box of person for 3DPW dataset?

**Comment by geopavlakos** [comment]
See how it's calculated [here](https://github.com/nkolot/SPIN/blob/2476c436013055be5cb3905e4e4ecfa86966fac3/datasets/preprocess/pw3d.py#L46) from ground truth 2D keypoint locations.

**Comment by thribhuvanrapolu** [comment]
Thanks for clarifying!
